File: SDKs/figshare.com/2.0.0/typescript/src/sdk/models/operations/privateinstitutionembargooptionsdetails.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PrivateInstitutionEmbargoOptionsDetailsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oAuth2: shared.SchemeOAuth2;
}


export class PrivateInstitutionEmbargoOptionsDetailsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  security: PrivateInstitutionEmbargoOptionsDetailsSecurity;
}


export class PrivateInstitutionEmbargoOptionsDetailsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorMessage?: shared.ErrorMessage;

  @SpeakeasyMetadata({ elemType: shared.GroupEmbargoOptions })
  groupEmbargoOptions?: shared.GroupEmbargoOptions[];

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/figshare.com/2.0.0/python/src/sdk/models/operations/private_institution_embargo_options_details.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import errormessage as shared_errormessage
from ..shared import groupembargooptions as shared_groupembargooptions


@dataclasses.dataclass
class PrivateInstitutionEmbargoOptionsDetailsSecurity:
    o_auth2: shared_security.SchemeOAuth2 = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class PrivateInstitutionEmbargoOptionsDetailsRequest:
    security: PrivateInstitutionEmbargoOptionsDetailsSecurity = dataclasses.field()
    

@dataclasses.dataclass
class PrivateInstitutionEmbargoOptionsDetailsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    error_message: Optional[shared_errormessage.ErrorMessage] = dataclasses.field(default=None)
    group_embargo_options: Optional[list[shared_groupembargooptions.GroupEmbargoOptions]] = dataclasses.field(default=None)
    
